main.cpp

```diff
@@ -19,13 +19,22 @@ using namespace clang::tooling;
 
 class CastCallBack : public MatchFinder::MatchCallback {
 public:
-    CastCallBack(Rewriter& rewriter) {
-        // Your code goes here
-    };
+    CastCallBack(Rewriter& rewriter) : rewriter(rewriter) {}
 
     void run(const MatchFinder::MatchResult &Result) override {
-        // Your code goes here
+        auto cStyleCExpr = Result.Nodes.getNodeAs<CStyleCastExpr>("cast");
+        if (cStyleCExpr) {
+            auto lPloc = cStyleCExpr->getLParenLoc();
+            auto rPloc = cStyleCExpr->getRParenLoc();
+            auto endLoc = cStyleCExpr->getEndLoc();
+
+            rewriter.ReplaceText(lPloc, 1, "static_cast<");
+            rewriter.ReplaceText(rPloc, 1, ">(");

```

Comment:
Please describe how to change the program code to get the following result:
```cpp
int main() {
    float fooooo;
    int i = static_cast<int>          (fooooo)      ;
    return 0;
}
```
 for the origin `test.cpp`:
```cpp
int main() {

    float fooooo;

    int i = (int)          fooooo      ;

    return 0;

}
```

Reply:
What you mean? Spaces? Sorry, I don't understand.

Reply:
If for the following example:
```cpp
int main() {
    float fooooo;
    int i = (int)          fooooo      ;
    return 0;
}
```
You run your current solution you will get the following result:
```cpp
int main() {
    float fooooo;
    int i = static_cast<int>(          fooooo)      ;
    return 0;
}
```
The question is how to change the code to get the following result:
```cpp
int main() {
    float fooooo;
    int i = static_cast<int>          (fooooo)      ;
    return 0;
}
```

Reply:
Change `rewriter.ReplaceText(rPloc, 1, ">(");' to `rewriter.ReplaceText(rPloc, 1, ">");`, after add command `rewriter.InsertText(cStyleCExpr->getSubExprAsWritten()->getBeginLoc(), "(");`.